dev-app/src/api/api.ts

```diff
@@ -141,17 +150,27 @@ export class Api {
       }
     }
 
-    // TODO: implement connect functionality to set these values
-    // if (
-    //   this.connectedAccount &&
-    //   this.connectedAccount.type === ConnectChargeType.Destination
-    // ) {
-    //   formData.append('on_behalf_of', this.connectedAccount.id);
-    //   formData.append('transfer_data[destination]', this.connectedAccount.id);
-    // }
+    if (
+      this.currentAccount &&
+      this.currentAccount.connectedAccountType === ChargeType.DestinationCharges
+    ) {
+      formData.append('on_behalf_of', this.currentAccount.stripeAccountID);
+      formData.append(
+        'transfer_data[destination]',
+        this.currentAccount.stripeAccountID
+      );
+      formData.append('application_fee_amount', '200');
```

Comment:
Application fee should be dynamic instead of hard-coded, passed in https://github.com/stripe/stripe-terminal-react-native/blob/6f7d6353/dev-app/src/screens/CollectCardPaymentScreen.tsx#L135-L141 again with inputValues.applicationFeeAmount